packages/wrangler/src/pages/dev.ts

```diff
@@ -497,9 +607,28 @@ export const Handler = async (args: PagesDevArguments) => {
 							getFunctionsBuildWarning(functionsDirectory, e.message)
 						);
 					} else {
-						throw e;
+						/*
+						 * don't break developer flow in watch mode by throwing an error
+						 * here. Many times errors will be just the result of unfinished
+						 * typing. Instead, log the error, point out we are still serving
+						 * the last successfully built Functions, and allow developers to
+						 * write their code to completion
+						 */
+						logger.warn(
```

Comment:
should this be an error?

Reply:
I feel like this should be an error? AFAIK it's a recoverable error (i.e. saving the file again should re-trigger the build), but it feels like we shouldn't hide build errors?

Reply:
yeah, what we'll see is smth like
```
[WARNING] Error while building Functions.....

<error details>
```

Reply:
converted to `logger.error` ✅